pkg/controller/controllerutil/controllerutil.go

```diff
@@ -32,6 +34,33 @@ import (
 	"sigs.k8s.io/controller-runtime/pkg/client/apiutil"
 )
 
+const inClusterNamespacePath = "/var/run/secrets/kubernetes.io/serviceaccount/namespace"
+
+var inClusterNamespace string
+
+// GetInClusterNamespace returns the namespace by which the controller is running under.
+func GetInClusterNamespace() (string, error) {
+	if inClusterNamespace != "" {
+		return inClusterNamespace, nil
+	}
+	// Check whether the namespace file exists.
+	// If not, we are not running in cluster so can't guess the namespace.
+	_, err := os.Stat(inClusterNamespacePath)
+	if os.IsNotExist(err) {
+		return "", fmt.Errorf("not running in-cluster")
+	} else if err != nil {
+		return "", fmt.Errorf("error checking namespace file: %w", err)
+	}
+
+	// Load the namespace file and return its content
+	namespace, err := ioutil.ReadFile(inClusterNamespacePath)
+	if err != nil {
+		return "", fmt.Errorf("error reading namespace file: %w", err)
+	}
+	inClusterNamespace = string(namespace)
+	return inClusterNamespace, nil
```

Comment:
```suggestion
	return string(namespace), nil
```

Reply:
That was a deliberate memoization (and pretty much the one thing I changed to the existing func), so we don't have to read the file on every call.

Reply:
Ah ok